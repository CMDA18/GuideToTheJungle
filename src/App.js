// @flow

import React, { type ComponentType } from 'react'
import Helmet from 'react-helmet'
import { Switch, Route } from 'react-router'

import { isClient } from './lib/side'
import routes, { type Route as RouteObjectType } from './routes'
import './styles/index'

const COMMIT_HASH = process.env.COMMIT_HASH || 'N.A.'

/* istanbul ignore next */
if (
  process.env.NODE_ENV !== 'test' &&
  isClient
) {
  console.info(`
  V:${COMMIT_HASH}

  `)
}

export const RouteWithSubRoutes = (route: RouteObjectType) => {
  return (
    <Route path={route.path} render={(props) => {
      /* istanbul ignore next */
      if (route.status && props.staticContext) {
        props.staticContext.status = route.status
      }
      /* istanbul ignore next */
      return <route.component {...props} />
    }} />
  )
}

const App: ComponentType<*> = () => (
  <div>
    <Helmet
      titleTemplate='%s - Basic Project'
      title='Shop'
    />
    <Switch>
      {
        routes.map((route, i) => {
          if (route.preload) { delete (route.preload) }
          return (
            <RouteWithSubRoutes key={i} {...route} />
          )
        })
      }
    </Switch>
  </div>
)

export default App
