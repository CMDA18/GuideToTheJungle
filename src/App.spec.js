import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import App, { RouteWithSubRoutes } from './App'

it('renders without crashing', () => {
  expect(toJson(shallow(<App/>))).toMatchSnapshot()
})

it('renders route', () => {
  expect(toJson(shallow(<RouteWithSubRoutes route={{path: '/', component: React.createElement('h1', { 'data-test': 'test' }, 'test')}} />))).toMatchSnapshot()
})
