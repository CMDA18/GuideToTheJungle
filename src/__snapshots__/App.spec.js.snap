// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders route 1`] = `
<Route
  render={[Function]}
/>
`;

exports[`renders without crashing 1`] = `
<div>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Shop"
    titleTemplate="%s - Basic Project"
  />
  <Switch>
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="0"
      path="/"
      title="Homepage"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="1"
      path="/redirectToHome"
      title="redirect to home"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="2"
      path="/overview"
      title="Overview"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="3"
      path="/faq"
      title="FAQ"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="4"
      path="/jungleMinds"
      title="1 Jungle Minds"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="5"
      path="/ourStory"
      title="Our Story"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="6"
      path="/jungleMaslow"
      title="Jungle Maslow"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="7"
      path="/finances"
      title="Finances"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="8"
      path="/tasksModuleOne"
      title="Tasks Module One"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="9"
      path="/jungleCulture"
      title="2 The Jungle Culture"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="10"
      path="/jungleFormula"
      title="2.1 Jungle Formula"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="11"
      path="/jungleDays"
      title="2.2 Jungle Days"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="12"
      path="/ownGrowth"
      title="2.3 Ontwikkeling"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="13"
      path="/tasksModuleTwo"
      title="2.4 Opdrachten"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="14"
      path="/ourPeople"
      title="3 Our People"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="15"
      path="/ourTeam"
      title="3.1 Onze Team"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="16"
      path="/theWallOfFame"
      title="3.2 The Wall Of Fame"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="17"
      path="/theGoToPeople"
      title="3.3 The Go To People"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="18"
      path="/tasksModuleThree"
      title="3.4 Opdrachten"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="19"
      path="/howWeWork"
      title="4 How We Work"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="20"
      path="/ourMethode"
      title="4.1 Onze Methode"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="21"
      path="/insideTheJungle"
      title="4.2 Binnen ons Jungle"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="22"
      path="/outsideTheJungle"
      title="4.3 Buiten de Jungle"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="23"
      path="/tasksModuleFour"
      title="4.4 Opdrachten"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="24"
      path="/ourClients"
      title="Our Clients"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="25"
      path="/clients"
      title="5.1 Klanten"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="26"
      path="/portfolio"
      title="5.2 Portfolio"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="27"
      path="/future"
      title="5.3 Toekomst"
    />
    <RouteWithSubRoutes
      component={[Function]}
      exact={true}
      key="28"
      path="/tasksModuleFive"
      title="5.4 Opdrachten"
    />
    <RouteWithSubRoutes
      component={[Function]}
      key="29"
      path="*"
      status={404}
      title="Page Not Found"
    />
  </Switch>
</div>
`;
